ssz_serialization/merkleization.nim

```diff
@@ -635,6 +721,248 @@ func hashTreeRootAux[T](x: T): Digest =
   else:
     unsupported T
 
+func hashTreeRootAux[T](
+    x: T,
+    indices: openArray[GeneralizedIndex],
+    roots: var openArray[Digest],
+    loopOrder: seq[int],
+    slice: Slice[int],
+    atLayer: int): Result[void, string] =
+  when T is BasicType:
+    for i in slice:
+      if indexAt(i) != 1.GeneralizedIndex: return unsupportedIndex
+      rootAt(i) = hashTreeRootAux(x)
+  elif T is BitArray:
+    ? hashTreeRootAux(x.bytes, indices, roots, loopOrder, slice, atLayer)
+  elif T is BitList:
+    const
+      totalChunks = maxChunksCount(T, x.maxLen)
+      firstChunkIndex = nextPow2(totalChunks.uint64)
+      chunkLayer = log2trunc(firstChunkIndex)
+    var i = slice.a
+    while i <= slice.b:
+      let
+        index = indexAt(i)
+        indexLayer = log2trunc(index)
+      if index == 1.GeneralizedIndex:
+        let contentsHash = bitListHashTreeRoot(totalChunks, BitSeq x)
+        rootAt(i) = mixInLength(contentsHash, x.len)
+        inc i
+      elif index == 3.GeneralizedIndex:
+        rootAt(i) = hashTreeRootAux(toSszType(x.len.uint64))
```

Comment:
Reason was consistency with the existing code above this hunk (~line 712):

```
      addField hashTreeRoot(toSszType(x.value))
```

I also don't understand why it's necessary there, figured it would be something weird like the dot-syntax not working in some non-testable scenarios, so did the same here.

It works fine for me without the `toSszType`, I have removed it here.